js/slideshow.js

```diff
@@ -184,6 +183,98 @@
 		},
 
 		/**
+		 * Determines which colour to use for the background
+		 *
+		 * @param {*} image
+		 *
+		 * @returns {string}
+		 * @private
+		 */
+		_getBackgroundColour: function (image) {
+			var backgroundColour = this.darkBackgroundColour;
+			if (this._isTransparent(image.mimeType) && this._isMainlyDark(image)) {
+				backgroundColour = this.lightBackgroundColour;
+			}
+			return backgroundColour;
+		},
+
+		/**
+		 * Calculates the luminance of an image to determine if an image is mainly dark
+		 *
+		 * @param {*} image
+		 *
+		 * @returns {boolean}
+		 * @private
+		 */
+		_isMainlyDark: function (image) {
+			var isMainlyDark = true;
+			var numberOfSamples = 1000; // Seems to be the sweet spot
+			// The name has to be 'canvas'
+			var lumiCanvas = document.createElement('canvas');
+
+			var imgArea = image.width * image.height;
+			var canArea = numberOfSamples;
+			var factor = Math.sqrt(canArea / imgArea);
+
+			var scaledWidth = factor * image.width;
+			var scaledHeight = factor * image.height;
+			lumiCanvas.width = scaledWidth;
+			lumiCanvas.height = scaledHeight;
+			var lumiCtx = lumiCanvas.getContext('2d');
+			lumiCtx.drawImage(image, 0, 0, scaledWidth, scaledHeight);
+			var imgData = lumiCtx.getImageData(0, 0, lumiCanvas.width, lumiCanvas.height);
+			var pix = imgData.data; // pix.length will be approximately 4*numberOfSamples (for RGBA)
+			var pixelArraySize = pix.length;
+			var totalLuminance = 0;
+			var sampleNumber = 1;
+			var averageLuminance;
+			var totalAlpha = 0;
+			var alphaLevel;
+			var red = 0;
+			var green = 0;
+			var blue = 0;
+			var alpha = 0;
+			var lum = 0;
+			var alphaThreshold = 0.1;
+
+			var sampleCounter = 0;
+			var itemsPerPixel = 4; // red, green, blue, alpha
+			var sampleSize = Math.floor(pixelArraySize / (itemsPerPixel * numberOfSamples));
```

Comment:
Since the `canvas` size is scaled to be approximately `numberOfSamples`, the `sampleSize` variable is not needed anymore and the whole pixel array can be looped over:

```js
for (var i = 0, n = pixelArraySize; i < n; i += itemsPerPixel) {
```